packages/mui-material/src/Radio/Radio.js

```diff
@@ -62,7 +63,7 @@ const RadioRoot = styled(SwitchBase, {
         },
       },
       ...Object.entries(theme.palette)
-        .filter(([, palette]) => palette && palette.main)
+        .filter(createSimplePaletteValueFilter())
```

Comment:
What's the impact on performance? Would it make more sense for the theme provider to process the theme and generate a list of compatible value once so it's cached?